packages/cdk8s/src/_loadurl.js

```diff
@@ -17,7 +17,16 @@ if (!url) {
 
 const purl = parse(url);
 
-if (!purl.protocol) {
+let isLocalFile = false;
+try {
+  if(fs.lstatSync(url).isFile()){
+    isLocalFile = true;
+  }
+} catch (error) {
+  isLocalFile = false;
+}
+
+if (isLocalFile) {
```

Comment:
If it's a local file, shouldn't we just pass `url` to `fs.createReadStream` below? and in that case we can remove the `if purl.pathname` check.

will `pathname` work the same?

The flow is a little confusing to me, can we first check if it's a local file, and if so just call `createReadStream`, otherwise, parse the url, validate we have a protocol, and call the `http` module.

WDYT?

Reply:
Hi, thanks for the feedback, the initial code was indeed pretty verbose so I've trimmed it and implemented the changes you requested. Please let me know if there's anything else I can do further to improve it.